### DISCUSSION TITLE
Setting data but not redrawing DOM?

### DISCUSSION THREAD

**Discussion by dochoss** [nuxt/nuxt#21642] [Questions] [answered]
Hey folks, quick reactivity question.  I'm using Vue 3 script setup format, Nuxt 3, and Tailwind CSS, and I'm having an issue getting this one element to draw and update properly.  When I load the component, some values are showing as undefined despite seeing them correctly in the Vue DevTools and at other places in the DOM.  Any help with what I'm doing wrong here?

Markup - status.color is returning undefined initially when the DOM is drawn, but the text (i.e. the value for `{{ status.name }}` is being updated correctly
```
<div class="ml-auto mr-2 my-auto">
  <dt class="sr-only">Status</dt>
  <dd
    :class="`bg-${status.color}-100 text-${status.color}-800 ring-${status.color}-600/20`"
    class="shadow-sm inline-flex items-center rounded-md px-2 py-1 text-xs font-medium ring-1 ring-inset"
  >
    {{ status.name }}
  </dd>
</div>
```

Script setup - statuses is an array of status objects with just name and color properties.  I've tried onBeforeMount and onMounted and both produce a badge without any colors:
```
const status = ref({});

onBeforeMount(() => {
  status.value = statuses[Math.floor(Math.random() * statuses.length)];
});
```

The DOM gets rendered with `bg-undefined-100 text-undefined-800 ring-undefined-600/20` but if I change the `{{ status.name }}` to `{{ status.color }}` I do see that the color is being updated correctly and I get the result `red` (as an example).  However, the DOM isn't being redrawn to reflect the updated values, so the text will say `red` but the DOM still says `bg-undefined-100`.  Any idea what I'm doing wrong?

Edit: I just noticed this warning appearing in the console:
![image](https://github.com/nuxt/nuxt/assets/17858793/815b6275-2d6f-4fe7-9639-aea6a0c7d425)

**Comment by maxarias-io** [comment]
This is more of a Vue question, the mounted hooks only run on the client. That's why you're seeing the hydration mismatch.

**Reply by dochoss** [reply]
Gotcha.  In my `script setup` I have an array of statuses with name and color properties.  What would be the ideal way to randomly choose one of those when the component is rendered?

**Reply by maxarias-io** [reply, marked as answer]
I've used `useAsyncData` before for random data, that way it's generated in the server and hydrated in the client.

**Reply by dochoss** [reply]
Sweet, thanks!  I'll give it a try
